scrapy/contracts/__init__.py

```diff
@@ -84,8 +86,11 @@ def cb_wrapper(response):
 
         def eb_wrapper(failure):
             case = _create_testcase(method, 'errback')
-            exc_info = failure.value, failure.type, failure.getTracebackObject()
-            results.addError(case, exc_info)
+            exc_info = failure.type, failure.value, failure.getTracebackObject()
+            if failure.type.__name__ in ignored_errors:
```

Comment:
Right. `ignored_errors` was changed from a set into a list in the last commit (126e6a2). It could be changed back into a set by changing line 58 of `scrapy/contracts/__init__.py`:
```py
ignored_errors.add(getattr(contract, 'ignored_errors'))
```